colcon_clean/subverb/__init__.py

```diff
@@ -259,13 +267,46 @@ def _onexc(func, path, excinfo):  # pragma: no cover
     raise
 
 
-def _clean_path(path):
+def _clean_path(path: Path):
     logger.info(f"Cleaning path: '{path}'")
     if path.is_dir():
         try:
             shutil.rmtree(path, onexc=_onexc)
         except TypeError:
             # TODO: Remove when minimum python version is 3.12
             shutil.rmtree(path, onerror=_onerror)
-    else:
+    elif path.exists():
         path.unlink()
+    else:
+        logger.warning(f"Path does not exist (anymore): '{path}'")
+
+
+def _filter_paths(paths: Iterable[Path]) -> List[Path]:
+    # Convert to absolute, normalized Path objects
+    paths = sorted(
+        {p.resolve() for p in paths}
+    )  # Sorted, so parents come before children
+
+    result = []
+    if sys.version_info >= (3, 9):
+        for path in paths:
+            if not any(path.is_relative_to(parent) for parent in result):
+                result.append(path)
+            else:
+                logger.debug(f"Skipping path: '{path}' as it is a child path")
```

Comment:
What's an example that hits this code path? A a symlink that point outside the colcon workspace? Could we add a test to cover these new lines?